src/learning/InitializeProfile.cpp

```diff
@@ -0,0 +1,203 @@
+    float delta) {
+  // creating imaginary profile performance for criterion crit
+  std::string critId = crit.getId();
+  float imaginaryProfilePerformance =
+      altPerformance_.getPerf(altId, critId).getValue() + delta;
+
+  // Creating 2 int that will count the number of correctly classified
+  // alternatives for criterion crit for a profile performance of
+  // imaginaryProfilePerformance.
+  int catAboveCounter = 0;
+  int catBelowCounter = 0;
+
+  // candidates are computed thanks to the method
+  // ProfileInitializer::getProfilePerformanceCandidates(crit, catBelow,
+  // nbCategories);
+  for (std::string can : candidates) {
+    // if the performance of the candidate is higher than the profile
+    // performance and that the alternatives category is catAbove then we have
+    // correctly classified it
+    if (altPerformance_.getPerf(can, critId).getValue() >
+            imaginaryProfilePerformance and
+        altPerformance_.getAlternativeAssignment(can) == catAbove) {
+      ++catAboveCounter;
+      // if the performance of the candidate is lower than the profile
+      // performance and that the alternatives category is catBelow then we have
+      // correctly classified it
+    } else if (altPerformance_.getPerf(can, critId).getValue() <
+                   imaginaryProfilePerformance and
+               altPerformance_.getAlternativeAssignment(can) == catBelow) {
+      ++catBelowCounter;
+    }
+  }
+  float proba;
+  proba = catAboveCounter / catFrequency[catAbove.getCategoryRank()] +
+          catBelowCounter / catFrequency[catBelow.getCategoryRank()];
+  return proba;
+}
+
+std::vector<Perf> ProfileInitializer::initializeProfilePerformance(
+    const Criterion &crit, Categories &categories,
+    const std::vector<float> &catFre) {
+  std::vector<float> categoryLimits;
+  int nbCategories = categories.getNumberCategories();
+
+  for (int i = 0; i < nbCategories - 1; i++) {
+    std::vector<float> altProba;
+    std::vector<std::string> candidates =
+        ProfileInitializer::getProfilePerformanceCandidates(crit, categories[i],
+                                                            nbCategories);
+    // OPTIM : POSSIBILITY parallelization synchrone
+    for (std::string cand : candidates) {
+      float proba = ProfileInitializer::weightedProbability(
+          cand, crit, categories[i], categories[i + 1], nbCategories, catFre,
+          candidates);
+      altProba.push_back(proba);
+    }
+    float totProba = std::accumulate(altProba.begin(), altProba.end(), 0);
+    float randomNumber = getRandomUniformFloat(0, 0, totProba);
+    float tmp = altProba[0];
+    int index;
+    for (int i = 1; i < candidates.size(); i++) {
+      if (tmp < randomNumber) {
+        tmp += altProba[i];
+      } else {
+        index = i - 1;
+        break;
+      }
+    }
+    categoryLimits.push_back(
+        altPerformance_.getPerf(candidates[index], crit.getId()).getValue());
```

Comment:
Same here, consider not using getPerf here, you might need to add a new function in perfTable to get all the Perf you need in one shot

Reply:
Same here do not understand

Reply:
See two comments above: getPerf is not efficient. We will probably need to find a way to not call those and just pass the Perf objects to the functions

Reply:
still don't get it ahah